### PR TITLE
869 improve error messages when connecting to a remote repository

### PR DESCRIPTION
**Pull request by swordEdge**

### CODE REVIEW

**Review by six7 — CHANGES_REQUESTED** [review]
I'd love it if we would also display `validation` errors, as in the contents failed to pass the validation checks we have in the various TokenStorage.ts files (e.g. in https://github.com/six7/figma-tokens/blob/32f8cdffe157812374059789000d8995b179848f/src/storage/FileTokenStorage.ts#L44) -> we should show the user then that their file contents don't pass the schema, e.g. `Contents don't pass schema validation`

File: src/storage/ADOTokenStorage.ts [hunk 1]
```diff
@@ -309,6 +311,9 @@ export class ADOTokenStorage extends GitTokenStorage {
           })),
         ];
       }
+      return {
+        errorMessage: ErrorMessages.VALIDATION_ERROR,
+      };
     } catch (e) {
```
Comment (six7) [inline src/storage/ADOTokenStorage.ts:317]: should we return an errormessage as well when an error is caught?

File: src/storage/RemoteTokenStorage.ts [hunk 2]
```diff
@@ -71,33 +76,43 @@ export abstract class RemoteTokenStorage<Metadata = unknown, SaveOptions = unkno
...
-        data.tokens = {
-          ...data.tokens,
-          ...parseTokenValues({ [file.name]: file.data }),
-        };
-      } else if (file.type === 'metadata') {
-        data.metadata = {
-          ...data.metadata,
-          ...file.data,
-        };
+    const content = await this.read();
```
Comment (six7) [inline src/storage/RemoteTokenStorage.ts:87]: should we keep `files` here? i feel it's easier to read + the singular is `file` later on as well

File: src/storage/__tests__/GithubTokenStorage.test.ts [hunk 3]
```diff
@@ -154,49 +155,66 @@ describe('GithubTokenStorage', () => {
     });
   });
 
-  it('can read from Git in single file format', async () => {
-    mockGetContent.mockImplementationOnce(() => (
+  // it('can read from Git in single file format', async () => {
```
Comment (six7) [inline src/storage/__tests__/GithubTokenStorage.test.ts:158]: i think we'll still want that test

**Review by six7 — CHANGES_REQUESTED** [review]
havent functionally tested this yet but went over the code

File: src/storage/ADOTokenStorage.ts (hunk 1 above)
Comment (swordEdge) [reply src/storage/ADOTokenStorage.ts:317]: This is the case when the content failed validation schema

Comment (swordEdge) [reply src/storage/ADOTokenStorage.ts:317]: This is the case when the content failed validation schema.

**Review by LiamMartens — APPROVED** [review]

**Review by six7 — APPROVED** [review]

**Pull request merged**